### PR TITLE
[Refactor] Decoupling missingUpdateEventsTranscoder

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Create a `PushNotificationStrategy`

### CODE REVIEW

File: Source/Analytics/NotificationsTracker.swift [hunk 1]
```diff
@@ -16,83 +16,83 @@
...
-        }
-    }
-}
-
-extension NotificationsTracker {
-    override public var debugDescription: String {
-        return "Current values: \(analytics?.persistedAttributes(for: eventName) ?? [:])"
-    }
-}
+//import Foundation
```
Comment (typfel) [inline Source/Analytics/NotificationsTracker.swift:19]: Delete the commented out code?

File: Source/Data Model/NSManagedObjectContext+LastNotificationID.swift [hunk 2]
```diff
@@ -16,37 +16,37 @@
...
-                else { return nil }
-            return uuid
-        }
-    }
-    
-    public var zm_hasLastNotificationID: Bool {
-        return zm_lastNotificationID != nil
-    }
-}
+//import Foundation
```
Comment (typfel) [inline Source/Data Model/NSManagedObjectContext+LastNotificationID.swift:19]: Delete the commented out code?

File: Source/Data Model/NSManagedObjectContext+ServerTimeDelta.swift [hunk 3]
```diff
@@ -16,26 +16,26 @@
...
-        
-        set {
-            precondition(zm_isSyncContext, "serverTimeDelta can only be accessed on the sync context")
-            userInfo[NSManagedObjectContext.ServerTimeDeltaKey] = newValue
-        }
-        
-    }
-        
-}
+//public extension NSManagedObjectContext {
```
Comment (typfel) [inline Source/Data Model/NSManagedObjectContext+ServerTimeDelta.swift:22]: Delete the commented out code?

File: Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.h [hunk 4]
```diff
@@ -17,55 +17,55 @@
...
-/// Returns an NSUUID to start with after calling resetFetching
-@optional
-- (NSUUID *)startUUID;
-
-/// Returns YES, if the error response for a specific statusCode should be parsed (e.g. if the payload contains content that needs to be processed)
-@optional
-- (BOOL)shouldParseErrorForResponse:(ZMTransportResponse*)response;
-
-@end
+//@import Foundation;
```
Comment (typfel) [inline Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.h:20]: Delete the commented out code?

File: Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.m [hunk 5]
```diff
@@ -17,141 +17,141 @@
...
-    id strongTranscoder = self.transcoder;
-    if ([strongTranscoder respondsToSelector:@selector(startUUID)]) {
-        self.lastUUIDOfPreviousPage = [strongTranscoder startUUID];
-    }
-    [self.singleRequestSync readyForNextRequest];
-}
-
-@end
-
+//@import WireSystem;
```
Comment (typfel) [inline Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.m:20]: Delete the commented out code?

File: Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m [hunk 6]
```diff
@@ -18,7 +18,8 @@
 
 
 @import WireTransport;
-@import WireSyncEngine;
+//@import WireSyncEngine;
```
Comment (typfel) [inline Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m:21]: Delete the commented out code?

File: Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m [hunk 7]
```diff
@@ -112,10 +113,10 @@ - (void)tearDown {
     [super tearDown];
 }
 
-@end
-
-
-@implementation ZMMissingUpdateEventsTranscoderTests (MissingNotifications)
+//@end
```
Comment (typfel) [inline Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m:116]: Delete the commented out code?

File: Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoder.m [hunk 8]
```diff
@@ -23,7 +23,7 @@
 
 #import "ZMMissingUpdateEventsTranscoder+Internal.h"
 #import <WireSyncEngine/WireSyncEngine-Swift.h>
-#import "ZMSimpleListRequestPaginator.h"
+//#import <WireRequestStrategy/ZMSimpleListRequestPaginator.h>
```
Comment (typfel) [inline Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoder.m:26]: Delete the commented out code?

File: Source/Analytics/NotificationsTracker.swift (hunk 1 above)
Comment (KaterinaWire) [reply Source/Analytics/NotificationsTracker.swift]: Done

File: Source/Data Model/NSManagedObjectContext+LastNotificationID.swift (hunk 2 above)
Comment (KaterinaWire) [reply Source/Data Model/NSManagedObjectContext+LastNotificationID.swift]: Done

File: Source/Data Model/NSManagedObjectContext+ServerTimeDelta.swift (hunk 3 above)
Comment (KaterinaWire) [reply Source/Data Model/NSManagedObjectContext+ServerTimeDelta.swift]: Done

File: Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.h (hunk 4 above)
Comment (KaterinaWire) [reply Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.h]: Done

File: Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.m (hunk 5 above)
Comment (KaterinaWire) [reply Source/Synchronization/Transcoders/Helper/ZMSimpleListRequestPaginator.m]: Done

File: Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoder.m (hunk 8 above)
Comment (KaterinaWire) [reply Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoder.m]: Done

File: Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m (hunk 7 above)
Comment (KaterinaWire) [reply Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m]: Done

File: Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m (hunk 6 above)
Comment (KaterinaWire) [reply Tests/Source/Synchronization/Transcoders/ZMMissingUpdateEventsTranscoderTests.m]: Done

File: Source/Calling/WireCallCenterV3+Notifications.swift [hunk 9]
```diff
@@ -112,15 +112,17 @@ public struct WireCallCenterMissedCallNotification : SelfPostingNotification {
 
 // MARK:- Received call observer
 
-public protocol WireCallCenterCallErrorObserver : class {
-    func callCenterDidReceiveCallError(_ error: CallError)
+public protocol WireCallCenterCallErrorObserver: class {
```
Comment (typfel) [inline Source/Calling/WireCallCenterV3+Notifications.swift:115]: These changes are not not from you? Is the base branch not update to date with develop?

Comment (KaterinaWire) [reply Source/Calling/WireCallCenterV3+Notifications.swift]: I believe that yes. Now the feature branch is updated.

**Pull request merged**